server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/materials/pluggable_scm_material_representer.rb

```diff
@@ -20,7 +20,12 @@ module Materials
             class PluggableScmMaterialRepresenter < ApiV1::BaseRepresenter
                 alias_method :material_config, :represented
 
-                property :scm_id, as: :ref
+                property :scmId, as: :ref, setter: lambda { |value, options|
+                  scm = options[:config].getSCMs().find(value)
+                  self.setSCMConfig(scm) if(scm)
+                  self.setScmId(value)
+                }
```

Comment:
@ketan - Well, the validation does happen at the server side and the response json would contain this error. The issue with throwing exceptions in a places like this is, it wouldn't try to validate the rest of the entity, so a user would get one error at a time. Thats frustrating from a user's point of view.